src/core/crypto/ed25519.ts

```diff
@@ -313,5 +325,23 @@ export class Ed25519Signature extends Signature {
     return new Ed25519Signature(bytes);
   }
 
+  /**
+   * Checks if an ED25519 signature is non-canonical.
+   */
+  checkSMalleability(): boolean {
```

Comment:
This is a bad name. I realize it is inherited from our Rust codebase.

I would rename this to `isCanonicalSignature`.